### PR TITLE
Add regex example to "Match detection for URIs" that is not broken

### PR DESCRIPTION
**Pull request by ypid**
I find it important that examples actually follow best practices. The current regex one is not for two reasons:

1. Not escaping dots in URIs when writing regex is a mistake that is often made, even by advanced regex people. I think this comes from the way we humans look at URIs. Dots are so common in there, aren’t they?
Yes, but we write proper regular expressions here, not something in between.

2. `^https://.*google\.com$` is an improper regex (as already pointed out because it also matches `malicious-site.com`) that is only there to show the weaknesses of regular expressions for this use case. I find such an example very good for this purpose but there should also be a "good example" that complements it.

I found such a "good example" that I hope is more useful and has no unwanted loopholes.

Updates: #41
cc: @gene1wood

### CODE REVIEW

**Comment by gene1wood** [comment]
@ypid Good catch on that mistake in the regex.

Would it be better to show a bad example and good example as you've done, or merely to fix the mistake in the regex (by escaping the `.` character)?

I wonder if the unfamiliar URL of `https://ap01.my.home/cgi-bin/luci/` might confuse readers.

**Comment by ypid** [comment]
> I wonder if the unfamiliar URL of https://ap01.my.home/cgi-bin/luci/ might confuse readers.

The URL is from OpenWrt. The `.home` TLD is suitable for home networks as mentioned in https://tools.ietf.org/html/rfc6762#appendix-G. But I can also change it to `int.example.net` for a RFC compliant, unbiased example.

**Comment by ypid** [comment]
> I wonder if the unfamiliar URL of https://ap01.my.home/cgi-bin/luci/ might confuse readers.

You are right, a more common URL should be used. I will update this PR if there is an interest in it.

**Comment by fschillingeriv** [comment]
@ypid @gene1wood Apologies, didn't intend for this to get closed by #393. Would absolutely welcome a "good example" to contrast the dangerous one, and agreed that the unfamiliar URL may escape some readers.

Perhaps a **safe** alternative to `^https://.*google\.com$` is the way to go?

**Comment by ypid** [comment]
Done, I try to avoid Google but I think I found a better example domain. Lets discuss in #394 if there is a need for discussion.

File: _articles/features/uri-match-detection.md [hunk 1]
```diff
@@ -75,11 +75,19 @@ Regular expressions are an advanced option and can be quite dangerous if used in
 
 Selecting **Regular expression** will prompt Bitwarden to offer auto-fill when the detected resources matches a specified [regular expression](https://en.wikipedia.org/wiki/Regular_expression){:target="_blank"}. Regular expressions are always *case insensitive*.
```
Comment (fschillingeriv) [inline _articles/features/uri-match-detection.md:77]:
```suggestion

#### Unsafe Example

```

File: _articles/features/uri-match-detection.md [hunk 2]
```diff
@@ -75,11 +75,19 @@ Regular expressions are an advanced option and can be quite dangerous if used in
...
 
-For example, if the URI vault `^https://.*google\.com$` uses regular expression match detection:
+Bad example, if the URI vault `^https://.*google\.com$` uses regular expression match detection:
```
Comment (fschillingeriv) [inline _articles/features/uri-match-detection.md:78]:
```suggestion
If the URI vault `^https://.*google\.com$` uses regular expression match detection:
```

File: _articles/features/uri-match-detection.md [hunk 3]
```diff
@@ -75,11 +75,19 @@ Regular expressions are an advanced option and can be quite dangerous if used in
...
 
 - **Auto-fill offered** for `https://google.com`, `https://sub.google.com`, `https://malicious-site.com?q=google.com`
 - **Auto-fill not offered** for `http://google.com` or `https://yahoo.com`
 
+This probably matches more than what is indented. Consider avoiding the `.` which matches any character entierly for this security sensitive use of regular expressions.
+
```
Comment (fschillingeriv) [inline _articles/features/uri-match-detection.md:84]:
```suggestion

#### Safe Example

```

File: _articles/features/uri-match-detection.md [hunk 4]
```diff
@@ -75,11 +75,19 @@ Regular expressions are an advanced option and can be quite dangerous if used in
...
+Good example, if the URI vault `^https://[a-z]+\.wikipedia\.org/w/index\.php` uses regular expression match detection:
```
Comment (fschillingeriv) [inline _articles/features/uri-match-detection.md:85]:
```suggestion
If the URI vault `^https://[a-z]+\.wikipedia\.org/w/index\.php` uses regular expression match detection:
```

**Comment by fschillingeriv** [comment]
@ypid apologies for the delay on this, we've had some great stuff happening at Bitwarden this month. Just made a few small suggestions. 

I think "bad example" could be interpreted as "incorrect", so I shifted the language to "safe"/"unsafe".

**Comment by ypid** [comment]
Sounds good to me. Done that.

**Review by fschillingeriv — APPROVED** [review]

**Pull request merged**